docs/gates.md

```diff
@@ -1,46 +1,224 @@
+The gate `cirq.Y**t` and the equivalent representation
+`cirq.YPowGate(exponent=t)` are measured in terms of 180 degree turns, while
+the function `cirq.ry(θ)` uses a radian angle of rotation.
+
+**cirq.X / cirq.XPowGate / cirq.rx** Rotations about the Pauli ``X`` axis.
+The gate `cirq.X**t` and the equivalent representation
+`cirq.XPowGate(exponent=t)` are measured in terms of 180 degree turns, while
+the function `cirq.rx(θ)` uses a radian angle of rotation.
+
+**cirq.PhasedXPowGate**
+This gate is a rotation about an axis in the XY plane of the Bloch sphere.
+The ``PhasedXPowGate`` takes two parameters, ``exponent`` and ``phase_exponent``.
+The gate is equivalent to the circuit `───Z^-p───X^t───Z^p───` where `p` is the
+`phase_exponent` and `t` is the `exponent`.
+
+**cirq.H / cirq.HPowGate** The Hadamard gate is a rotation around the X+Z axis.
+``cirq.HPowGate(exponent=t)`` is a variable rotation of t turns around this
+axis. ```cirq.H` is a π rotation and is equivalent to
+``cirq.HPowGate(exponent=1)``
+
+**S** The square root of Z gate, equivalent to ``cirq.Z ** 0.5``
+
+**T** The fourth root of Z gate, equivalent to ``cirq.Z ** 0.25``.  Often used
+in error correction.
+
+
+## Two qubit gates
+
+**cirq.CZ / cirq.CZPowGate** The controlled-Z gate.  A two qubit gate that
+phases the |11> state.  `cirq.CZPowGate(exponent=y)` is equivalent to
+`cirq.CZ**t` and has a matrix representation of ``exp(i pi |11><11| t)``.
 
-A class that implements ``Gate`` can be applied to qubits to produce an ``Operation``.
-In order to support functionality beyond that basic task, it is necessary to implement several *magic methods*.
+**cirq.CNOT / cirq.CNotPowGate** The controlled-X gate.  This gate swaps the
+|11⟩ and |10⟩ states.  ``cirq.CNotPowGate(exponent=t)`` is equivalent
+to ``cirq.CNOT ** t`` .
 
-Standard magic methods in python are `__add__`, `__eq__`, and `__len__`.
-Cirq defines several additional magic methods, for functionality such as parameterization, diagramming, and simulation.
-For example, if a gate specifies a `_unitary_` method that returns a matrix for the gate, then simulators will be able to simulate applying the gate.
-Or, if a gate specifies a `__pow__` method that works for an exponent of -1, then `cirq.inverse` will start to work on lists including the gate.
+**cirq.SWAP / cirq.SwapPowGate** The swap gate swaps the |01⟩ and |10⟩ states.
+``cirq.SWAP ** t`` is the same as ``cirq.SwapPowGate(exponent = t)``
 
-We describe some magic methods below.
+**cirq.ISWAP / cirq.ISwapPowGate**  The swap gate swaps the |01⟩ and |10⟩
+states and adds a relative phase of i.  ``cirq.ISWAP ** t`` is the same as
+``cirq.ISwapPowGate(exponent = t)``
 
+**Parity gates**: The gates cirq.XX, cirq.YY, and cirq.ZZ are equivalent to
+performing the equivalent one-qubit Pauli gates on both qubits.  The gates
+cirq.XXPowGate, cirq.YYPowGate, and cirq.ZZPowGate are the exponentiations of
```

Comment:
I think "power" is more accurate than "exponentiation", which to me sounds ambiguous as to whether the gate is the base or the exponent.

Reply:
We should have an explanation of what it means to raise a gate to a power, but that doesn't need to be in this PR.

Reply:
Changed all instances of exponentiations to powers.